File: CS/MapBubbleCharts/Form1.cs
using DevExpress.XtraMap;
using System.Data;
using System.Globalization;
using System.Windows.Forms;

namespace MapBubbleCharts {
    public partial class Form1 : Form {
        string dataPath = @"..\..\Earthquakes.xml";
        int minMagnitude = 6;
        int maxMagnitude = 10;

        VectorItemsLayer BubbleLayer { get { return (VectorItemsLayer)mapControl1.Layers["BubbleLayer"]; } }

        public Form1() {
            InitializeComponent();

            #region #DataProperty
            // Specify data for the bubble layer.
            BubbleLayer.Data = CreateData();
            #endregion #DataProperty
        }

        #region #CreateData
        private IMapDataAdapter CreateData() {
            BubbleChartDataAdapter adapter = new BubbleChartDataAdapter() {
                DataSource = LoadData(),
                MeasureRules = CreateMeasureRules(),
                ItemMaxSize = 60,
                ItemMinSize = 10
            };

            #region #Mappings
            // Map the properties of chart items to the appropriate fields in the data source.
            adapter.Mappings.Latitude = "glat";
            adapter.Mappings.Longitude = "glon";
            adapter.Mappings.Value = "mag";
            #endregion #Mappings

            // Create attribute mappings to provide additional information about map items to the map.
            adapter.AttributeMappings.Add(new MapItemAttributeMapping("Magnitude", "mag"));
            adapter.AttributeMappings.Add(new MapItemAttributeMapping("Year", "yr"));
            adapter.AttributeMappings.Add(new MapItemAttributeMapping("Month", "mon"));
            adapter.AttributeMappings.Add(new MapItemAttributeMapping("Day", "day"));
            adapter.AttributeMappings.Add(new MapItemAttributeMapping("Depth", "dep"));

            return adapter;
        }

        private DataTable LoadData() {
            DataSet ds = new DataSet();
            ds.ReadXml(dataPath);
            DataTable dt = ds.Tables[0];
            dt.DefaultView.RowFilter = string.Format(CultureInfo.InvariantCulture, "(mag >= {0}) AND (mag <= {1})", minMagnitude, maxMagnitude);

            return ds.Tables[0];
        }

        private MeasureRules CreateMeasureRules() {
            MeasureRules measureRules = new MeasureRules();

            measureRules.ApproximateValues = true;
            measureRules.RangeDistribution = new LinearRangeDistribution();

            for (int i = minMagnitude; i < maxMagnitude; i++) {
                measureRules.RangeStops.Add(i);
            }

            return measureRules;
        }
        #endregion #CreateData
    }
}

File: VB/MapBubbleCharts/Form1.vb
Imports DevExpress.XtraMap
Imports System.Data
Imports System.Windows.Forms

Namespace MapBubbleCharts
    Partial Public Class Form1
        Inherits Form

        Private dataPath As String = "..\..\Earthquakes.xml"
        Private minMagnitude As Integer = 6
        Private maxMagnitude As Integer = 10

        Private ReadOnly Property BubbleLayer() As VectorItemsLayer
            Get
                Return CType(mapControl1.Layers(1), VectorItemsLayer)
            End Get
        End Property

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            BubbleLayer.Data = CreateData()

            ' Create a size legend and define meanings for bubble sizes.
            mapControl1.Legends.Add(New SizeLegend() With {.Layer = BubbleLayer, .Header = "Magnitude"})
        End Sub

        Private Function CreateData() As IMapDataAdapter
            Dim adapter As New BubbleChartDataAdapter() With {.DataSource = LoadData()}

            ' Map the properties of chart items to the appropriate fields in the data source.
            adapter.Mappings.Latitude = "glat"
            adapter.Mappings.Longitude = "glon"
            adapter.Mappings.BubbleGroup = "Type"
            adapter.Mappings.Value = "mag"

            adapter.MeasureRules = CreateMeasureRules()

            ' Customize other chart adapter properties.
            adapter.ItemMaxSize = 60
            adapter.ItemMinSize = 10

            Return adapter
        End Function

        Private Function LoadData() As DataTable
            Dim ds As New DataSet()
            ds.ReadXml(dataPath)
            Dim dt As DataTable = ds.Tables(0)
            dt.DefaultView.RowFilter = String.Format("(mag >= {0}) AND (mag <= {1})", minMagnitude, maxMagnitude)

            Return ds.Tables(0)
        End Function

        Private Function CreateMeasureRules() As MeasureRules
            Dim measureRules As New MeasureRules()

            measureRules.ApproximateValues = True
            measureRules.RangeDistribution = New LinearRangeDistribution()

            For i As Integer = minMagnitude To maxMagnitude - 1
                measureRules.RangeStops.Add(i)
            Next i

            Return measureRules
        End Function

    End Class
End Namespace
